mesonbuild/interpreter/interpreter.py

```diff
@@ -1005,17 +1004,16 @@ def _do_subproject_meson(self, subp_name: str, subdir: str,
         self.build.subprojects[subp_name] = subi.project_version
         return self.subprojects[subp_name]
 
-    def _do_subproject_cmake(self, subp_name: str, subdir: str, subdir_abs: str,
+    def _do_subproject_cmake(self, subp_name: str, subdir: str,
                              default_options: T.Dict[OptionKey, str],
                              kwargs: kwtypes.DoSubproject) -> SubprojectHolder:
         with mlog.nested(subp_name):
-            new_build = self.build.copy()
```

Comment:
"Remove useless arguments".

Do you know why the original decision to make this use a copied self.build, was made? (I have no clue. It just looks very intentional, which makes me very curious.)

Reply:
Probably leftover of old refactoring. `_do_subproject_meson()` already pass a copy of self.build to the subproject's interpreter. The copy we pass to CMakeInterpreter is never used.